--- ispell-3.2.06/languages/american/Makefile	Sun Mar  2 00:15:57 2003
+++ ispell-3.2.06.new/languages/american/Makefile	Sun Mar  2 08:02:59 2003
@@ -366,14 +366,10 @@
   $(CBUILD)american.0 $(CBUILD)american.1 $(CBUILD)american.2 \
   $(CBUILD)altamer.0 $(CBUILD)altamer.1 $(CBUILD)altamer.2 \
   $(CBUILD)british.0 $(CBUILD)british.1 $(CBUILD)british.2:
-	rm -f english.[0-3] american.[0-2] altamer.[012] british.[012]
-	@set $(SHELLDEBUG); \
-	for i in english.0 english.1 english.2 english.3 \
-	  american.0 american.1 american.2 altamer.0 altamer.1 altamer.2 \
-	  british.0 british.1 british.2; do \
-	    ln -s $(DICTSRC)/$$i .  ||  ln $(DICTSRC)/$$i . \
-	      ||  cp $(DICTSRC)/$$i .; \
-	done
+	rm -f $(subst $(CBUILD),,$@)
+	@set $(SHELLDEBUG)
+	ln -s $(DICTSRC)/$(subst $(CBUILD),,$@) .  ||  ln $(DICTSRC)/$(subst $(CBUILD),,$@) . \
+	      ||  cp $(DICTSRC)/$(subst $(CBUILD),,$@) .
 
 clean:
 	rm -f core *.hash *.stat *.cnt
--- ispell-3.2.06/Makefile	Sun Mar  2 00:15:57 2003
+++ ispell-3.2.06.new/Makefile	Sun Mar  2 08:13:46 2003
@@ -237,11 +237,13 @@
 all-languages:	munchable
 	$(MAKE) LANGUAGE_TARGET=all SHELLDEBUG=$(SHELLDEBUG) language-subdirs
 
-install: config.sh all install-basic install-deformatters
+install: config.sh all touchfile install-deformatters
 install: install-dictbuild install-languages
 
 partial-install: config.sh all install-basic install-languages
 
+touchfile: install-basic
+
 install-basic:
 	@. ./config.sh; \
 	  set -x; \
@@ -274,11 +276,12 @@
 	  chmod 644 ispell$$MAN1EXT; \
 	  cd $$MAN4DIR; \
 	  chmod 644 ispell$$MAN4EXT
+	touch touchfile
 
-install-deformatters:
+install-deformatters: touchfile 
 	cd deformatters; $(MAKE) install
 
-install-dictbuild:
+install-dictbuild: touchfile
 	@. ./config.sh; \
 	  set -x; \
 	  [ -d $$BINDIR ]  ||  (mkdir $$BINDIR; chmod 755 $$BINDIR); \
@@ -314,7 +317,7 @@
 	  chmod 644 buildhash$$MAN1EXT \
 	    munchlist$$MAN1EXT findaffix$$MAN1EXT tryaffix$$MAN1EXT
 
-install-languages:
+install-languages: touchfile
 	$(MAKE) LANGUAGE_TARGET=install SHELLDEBUG=$(SHELLDEBUG) \
 	  language-subdirs
 	. ./config.sh; \
