--- ispell-3.3.02/Makefile.orig	2005-05-25 15:57:02.000000000 +0200
+++ ispell-3.3.02/Makefile	2023-12-16 21:06:36.230691971 +0100
@@ -307,7 +307,6 @@ install-basic:
 	@. ./config.sh; \
 	  set -x; \
 	  cd $$BINDIR; \
-	  strip ispell$$EXEEXT; \
 	  chmod 755 ispell$$EXEEXT
 	@. ./config.sh; \
 	  set -x; \
@@ -362,7 +361,6 @@ install-dictbuild:
 	@. ./config.sh; \
 	  set -x; \
 	  cd $$BINDIR; \
-	  strip buildhash$$EXEEXT icombine$$EXEEXT ijoin$$EXEEXT; \
 	  chmod 755 buildhash$$EXEEXT icombine$$EXEEXT ijoin$$EXEEXT \
 	    munchlist findaffix tryaffix
 	@. ./config.sh; \
