diff -ur ispell-3.1.old/munchlist.X ispell-3.1/munchlist.X
--- ispell-3.1.old/munchlist.X	Mon Jan 23 19:28:27 1995
+++ ispell-3.1/munchlist.X	Thu Sep 16 15:45:24 1999
@@ -1,4 +1,4 @@
-: Use /bin/sh
+#!/bin/sh
 #
 # $Id$
 #
@@ -128,7 +128,7 @@
     set -vx
 fi
 LIBDIR=!!LIBDIR!!
-TDIR=${TMPDIR-/usr/tmp}
+TDIR=${TMPDIR-/tmp}
 TMP=${TDIR}/munch$$
 SORTTMP="-T ${TDIR}"			# !!SORTTMP!!
 if [ -r ./icombine ]
