diff -Nur ispell-3.2.06.orig/Makefile ispell-3.2.06/Makefile
--- ispell-3.2.06.orig/Makefile	Wed Jul 25 23:51:45 2001
+++ ispell-3.2.06/Makefile	Thu Dec 26 23:13:00 2002
@@ -323,7 +323,7 @@
 	  cd $$LIBDIR; \
 	  if [ $$MASTERHASH != $$DEFHASH ]; then \
 	    rm -f $$DEFHASH; \
-	    ln $$MASTERHASH $$DEFHASH; \
+	    ln -s $$MASTERHASH $$DEFHASH; \
 	  fi
 
 munchable:	findaffix tryaffix munchlist buildhash ispell icombine
@@ -499,7 +499,7 @@
 	    :; \
 	  else \
 	    set -x; \
-	    rm -f msgs.h; ln $$msgs msgs.h  ||  cp $$msgs msgs.h; \
+	    rm -f msgs.h; ln -s $$msgs msgs.h  ||  cp $$msgs msgs.h; \
 	  fi
 
 FRC:
diff -Nur ispell-3.2.06.orig/ispell.1X ispell-3.2.06/ispell.1X
--- ispell-3.2.06.orig/ispell.1X	Wed Jul 25 23:51:46 2001
+++ ispell-3.2.06/ispell.1X	Thu Dec 26 23:14:31 2002
@@ -1475,6 +1475,17 @@
 Default dictionary to use, if no
 .B \-d
 flag is given.
+.IP CHARSET
+Only read if
+.B DICTIONARY
+is set. Default formatter type or character encoding to use, if no
+.B \-T
+or
+.B \-t
+or
+.B \-n
+flag is given.
+Useful if formatter type is recognized in affix-file.
 .IP WORDLIST
 Personal dictionary file name
 .IP INCLUDE_STRING
diff -Nur ispell-3.2.06.orig/ispell.c ispell-3.2.06/ispell.c
--- ispell-3.2.06.orig/ispell.c	Wed Jul 25 23:51:46 2001
+++ ispell-3.2.06/ispell.c	Thu Dec 26 23:22:28 2002
@@ -209,6 +209,7 @@
 #include <fcntl.h>
 #endif /* NO_FCNTL_H */
 #include <sys/stat.h>
+#include <unistd.h>
 
 static void	usage P ((void));
 static void	initckch P ((char * wchars));
@@ -331,6 +332,7 @@
     static char	outbuf[BUFSIZ];
     int		argno;
     int		arglen;
+    int		dictdeft = 0;
 
     Cmd = *argv;
 
@@ -339,6 +341,7 @@
     p = getenv ("DICTIONARY");
     if (p != NULL)
 	{
+	dictdeft++;
 	if (index (p, '/') != NULL)
 	    (void) strcpy (hashname, p);
 	else
@@ -355,6 +358,7 @@
 	p = rindex (libdictname, '.');
 	if (p != NULL)
 	    *p = '\0';
+	preftype = getenv ("CHARSET");
 	}
     else
 	(void) sprintf (hashname, "%s/%s", LIBDIR, DEFHASH);
@@ -552,7 +556,7 @@
 		    usage ();
 		tflag = DEFORMAT_NROFF;	/* nroff/troff mode */
 		deftflag = DEFORMAT_NROFF;
-		if (preftype == NULL)
+		if (preftype == NULL || dictdeft)
 		    preftype = "nroff";
 		break;
 	    case 't':			/* TeX mode */
@@ -560,7 +564,7 @@
 		    usage ();
 		tflag = DEFORMAT_TEX;
 		deftflag = DEFORMAT_TEX;
-		if (preftype == NULL)
+		if (preftype == NULL || dictdeft)
 		    preftype = "tex";
 		break;
 	    case 'H':			/* HTML mode */
@@ -568,7 +572,7 @@
 		    usage ();
 		tflag = DEFORMAT_SGML;	/* non-TeX mode */
 		deftflag = DEFORMAT_SGML;
-		if (preftype == NULL)
+		if (preftype == NULL || dictdeft)
 		    preftype = "sgml";
 		break;
 	    case 'k':			/* Set keyword tables */
diff -Nur ispell-3.2.06.orig/languages/altamer/Makefile ispell-3.2.06/languages/altamer/Makefile
--- ispell-3.2.06.orig/languages/altamer/Makefile	Wed Jul 25 23:51:48 2001
+++ ispell-3.2.06/languages/altamer/Makefile	Thu Dec 26 23:23:55 2002
@@ -158,7 +158,7 @@
 	  cd $$LIBDIR; \
 	  chmod 644 english.aff $(HASHFILES); \
 	  for i in $(HASHFILES); do \
-	    ln $$i $(LANGUAGE).hash; \
+	    ln -s $$i $(LANGUAGE).hash; \
 	    break; \
 	  done
 	@. $(CONFIG); \
@@ -246,7 +246,7 @@
 
 $(DBUILD)altamer.sml:	$(DBUILD)english.sml
 	rm -f altamer.sml
-	ln english.sml altamer.sml
+	ln -s english.sml altamer.sml
 
 $(DBUILD)english.sml:	$(CONFIG)
 $(DBUILD)english.sml:	english.0
@@ -259,7 +259,7 @@
 
 $(DBUILD)altamer.sml+:	$(DBUILD)english.sml+
 	rm -f altamer.sml+
-	ln english.sml+ altamer.sml+
+	ln -s english.sml+ altamer.sml+
 
 $(DBUILD)english.sml+:	$(CONFIG)
 $(DBUILD)english.sml+:	english.0
@@ -272,7 +272,7 @@
 
 $(DBUILD)altamer.med:	$(DBUILD)english.med
 	rm -f altamer.med
-	ln english.med altamer.med
+	ln -s english.med altamer.med
 
 $(DBUILD)english.med:	$(CONFIG)
 $(DBUILD)english.med:	english.0
@@ -285,7 +285,7 @@
 
 $(DBUILD)altamer.med+:	$(DBUILD)english.med+
 	rm -f altamer.med+
-	ln english.med+ altamer.med+
+	ln -s english.med+ altamer.med+
 
 $(DBUILD)english.med+:	$(CONFIG)
 $(DBUILD)english.med+:	english.0
@@ -298,7 +298,7 @@
 
 $(DBUILD)altamer.lrg:	$(DBUILD)english.lrg
 	rm -f altamer.lrg
-	ln english.lrg altamer.lrg
+	ln -s english.lrg altamer.lrg
 
 $(DBUILD)english.lrg:	$(CONFIG)
 $(DBUILD)english.lrg:	english.0
@@ -311,7 +311,7 @@
 
 $(DBUILD)altamer.lrg+:	$(DBUILD)english.lrg+
 	rm -f altamer.lrg+
-	ln english.lrg+ altamer.lrg+
+	ln -s english.lrg+ altamer.lrg+
 
 $(DBUILD)english.lrg+:	$(CONFIG)
 $(DBUILD)english.lrg+:	english.0
@@ -324,7 +324,7 @@
 
 $(DBUILD)altamer.xlg:	$(DBUILD)english.xlg
 	rm -f altamer.xlg
-	ln english.xlg altamer.xlg
+	ln -s english.xlg altamer.xlg
 
 $(DBUILD)english.xlg:	$(CONFIG)
 $(DBUILD)english.xlg:	english.0
@@ -337,7 +337,7 @@
 
 $(DBUILD)altamer.xlg+:	$(DBUILD)english.xlg+
 	rm -f altamer.xlg+
-	ln english.xlg+ altamer.xlg+
+	ln -s english.xlg+ altamer.xlg+
 
 $(DBUILD)english.xlg+:	$(CONFIG)
 $(DBUILD)english.xlg+:	english.0
diff -Nur ispell-3.2.06.orig/languages/american/Makefile ispell-3.2.06/languages/american/Makefile
--- ispell-3.2.06.orig/languages/american/Makefile	Wed Jul 25 23:51:47 2001
+++ ispell-3.2.06/languages/american/Makefile	Thu Dec 26 23:24:31 2002
@@ -161,7 +161,7 @@
 	  cd $$LIBDIR; \
 	  chmod 644 english.aff $(HASHFILES); \
 	  for i in $(HASHFILES); do \
-	    ln $$i $(LANGUAGE).hash; \
+	    ln -s $$i $(LANGUAGE).hash; \
 	    break; \
 	  done
 	@. $(CONFIG); \
@@ -249,7 +249,7 @@
 
 $(DBUILD)american.sml:	$(DBUILD)english.sml
 	rm -f american.sml
-	ln english.sml american.sml
+	ln -s english.sml american.sml
 
 $(DBUILD)english.sml:	$(CONFIG)
 $(DBUILD)english.sml:	english.0
@@ -261,7 +261,7 @@
 
 $(DBUILD)american.sml+:	$(DBUILD)english.sml+
 	rm -f american.sml+
-	ln english.sml+ american.sml+
+	ln -s english.sml+ american.sml+
 
 $(DBUILD)english.sml+:	$(CONFIG)
 $(DBUILD)english.sml+:	english.0
@@ -273,7 +273,7 @@
 
 $(DBUILD)american.med:	$(DBUILD)english.med
 	rm -f american.med
-	ln english.med american.med
+	ln -s english.med american.med
 
 $(DBUILD)english.med:	$(CONFIG)
 $(DBUILD)english.med:	english.0
@@ -285,7 +285,7 @@
 
 $(DBUILD)american.med+:	$(DBUILD)english.med+
 	rm -f american.med+
-	ln english.med+ american.med+
+	ln -s english.med+ american.med+
 
 $(DBUILD)english.med+:	$(CONFIG)
 $(DBUILD)english.med+:	english.0
@@ -297,7 +297,7 @@
 
 $(DBUILD)american.lrg:	$(DBUILD)english.lrg
 	rm -f american.lrg
-	ln english.lrg american.lrg
+	ln -s english.lrg american.lrg
 
 $(DBUILD)english.lrg:	$(CONFIG)
 $(DBUILD)english.lrg:	english.0
@@ -309,7 +309,7 @@
 
 $(DBUILD)american.lrg+:	$(DBUILD)english.lrg+
 	rm -f american.lrg+
-	ln english.lrg+ american.lrg+
+	ln -s english.lrg+ american.lrg+
 
 $(DBUILD)english.lrg+:	$(CONFIG)
 $(DBUILD)english.lrg+:	english.0
@@ -321,7 +321,7 @@
 
 $(DBUILD)american.xlg:	$(DBUILD)english.xlg
 	rm -f american.xlg
-	ln english.xlg american.xlg
+	ln -s english.xlg american.xlg
 
 $(DBUILD)english.xlg:	$(CONFIG)
 $(DBUILD)english.xlg:	english.0
@@ -333,7 +333,7 @@
 
 $(DBUILD)american.xlg+:	$(DBUILD)english.xlg+
 	rm -f american.xlg+
-	ln english.xlg+ american.xlg+
+	ln -s english.xlg+ american.xlg+
 
 $(DBUILD)english.xlg+:	$(CONFIG)
 $(DBUILD)english.xlg+:	english.0
diff -Nur ispell-3.2.06.orig/languages/british/Makefile ispell-3.2.06/languages/british/Makefile
--- ispell-3.2.06.orig/languages/british/Makefile	Wed Jul 25 23:51:48 2001
+++ ispell-3.2.06/languages/british/Makefile	Thu Dec 26 23:25:11 2002
@@ -161,7 +161,7 @@
 	  cd $$LIBDIR; \
 	  chmod 644 english.aff $(HASHFILES); \
 	  for i in $(HASHFILES); do \
-	    ln $$i $(LANGUAGE).hash; \
+	    ln -s $$i $(LANGUAGE).hash; \
 	    break; \
 	  done
 	@. $(CONFIG); \
@@ -249,7 +249,7 @@
 
 $(DBUILD)british.sml:	$(DBUILD)english.sml
 	rm -f british.sml
-	ln english.sml british.sml
+	ln -s english.sml british.sml
 
 $(DBUILD)english.sml:	$(CONFIG)
 $(DBUILD)english.sml:	english.0
@@ -261,7 +261,7 @@
 
 $(DBUILD)british.sml+:	$(DBUILD)english.sml+
 	rm -f british.sml+
-	ln english.sml+ british.sml+
+	ln -s english.sml+ british.sml+
 
 $(DBUILD)english.sml+:	$(CONFIG)
 $(DBUILD)english.sml+:	english.0
@@ -273,7 +273,7 @@
 
 $(DBUILD)british.med:	$(DBUILD)english.med
 	rm -f british.med
-	ln english.med british.med
+	ln -s english.med british.med
 
 $(DBUILD)english.med:	$(CONFIG)
 $(DBUILD)english.med:	english.0
@@ -285,7 +285,7 @@
 
 $(DBUILD)british.med+:	$(DBUILD)english.med+
 	rm -f british.med+
-	ln english.med+ british.med+
+	ln -s english.med+ british.med+
 
 $(DBUILD)english.med+:	$(CONFIG)
 $(DBUILD)english.med+:	english.0
@@ -297,7 +297,7 @@
 
 $(DBUILD)british.lrg:	$(DBUILD)english.lrg
 	rm -f british.lrg
-	ln english.lrg british.lrg
+	ln -s english.lrg british.lrg
 
 $(DBUILD)english.lrg:	$(CONFIG)
 $(DBUILD)english.lrg:	english.0
@@ -309,7 +309,7 @@
 
 $(DBUILD)british.lrg+:	$(DBUILD)english.lrg+
 	rm -f british.lrg+
-	ln english.lrg+ british.lrg+
+	ln -s english.lrg+ british.lrg+
 
 $(DBUILD)english.lrg+:	$(CONFIG)
 $(DBUILD)english.lrg+:	english.0
@@ -321,7 +321,7 @@
 
 $(DBUILD)british.xlg:	$(DBUILD)english.xlg
 	rm -f british.xlg
-	ln english.xlg british.xlg
+	ln -s english.xlg british.xlg
 
 $(DBUILD)english.xlg:	$(CONFIG)
 $(DBUILD)english.xlg:	english.0
@@ -333,7 +333,7 @@
 
 $(DBUILD)british.xlg+:	$(DBUILD)english.xlg+
 	rm -f british.xlg+
-	ln english.xlg+ british.xlg+
+	ln -s english.xlg+ british.xlg+
 
 $(DBUILD)english.xlg+:	$(CONFIG)
 $(DBUILD)english.xlg+:	english.0
diff -Nur ispell-3.2.06.orig/languages/deutsch/Makefile ispell-3.2.06/languages/deutsch/Makefile
--- ispell-3.2.06.orig/languages/deutsch/Makefile	Wed Jul 25 23:51:47 2001
+++ ispell-3.2.06/languages/deutsch/Makefile	Thu Dec 26 23:41:37 2002
@@ -107,8 +107,31 @@
 # If you change DICTOPTIONS in your local.h file, you will have to do
 # "make dictclean" to clear out the old dictionary before you re-make.
 #
-DICTALWAYS	=	adjektive.txt worte.txt verben.txt
-DICTOPTIONS	=	Use_LANGUAGES_from_config.X
+DICTALWAYS	= worte.txt \
+		  verben.txt \
+		  adjektive.txt \
+		  klein.txt \
+		  imperat.txt \
+		  abkuerz.txt \
+		  vornamen.txt \
+		  geographie.txt \
+		  latein.txt \
+		  oesterreich.txt \
+		  informatik.txt \
+		  infoabk.txt \
+		  elektronik.txt \
+		  alphabeta.txt \
+		  roemisch.txt \
+		  orgabk.txt \
+		  marken.txt
+
+DICTXLARGE	= worte2.txt \
+		  zusammen.txt \
+		  technik.txt \
+		  compeng.txt \
+		  geogra2.txt \
+		  vornam2.txt \
+		  namen.txt
 
 #
 # The following variables may be overridden by the superior Makefile,
@@ -116,7 +139,7 @@
 # the affix file is closely related to the dictionary chosen; don't
 # change the affix file unless you know what you are doing!
 #
-AFFIXES	=	deutsch.aff
+AFFIXES	=	deutsch.aff-hk
 
 #
 #	Set this to "-vx" in the make command line if you need to
@@ -124,39 +147,54 @@
 #
 SHELLDEBUG = +vx
 
-all:	deutsch.hash
+all:	deutschmed.hash deutschlxg.hash
 
 install: all $(CONFIG)
 	@. $(CONFIG); \
 	  set -x; \
 	  [ -d $$LIBDIR ]  ||  (mkdir $$LIBDIR; chmod 755 $$LIBDIR); \
-	  cd $$LIBDIR; rm -f deutsch.aff deutsch.hash
+	  cd $$LIBDIR; rm -f deutsch.aff deutschmed.hash deutschlxg.hash
 	@. $(CONFIG); \
 	  set -x; \
-	  cp deutsch.aff deutsch.hash $$LIBDIR
+	  cp deutsch.aff deutschmed.hash deutschlxg.hash $$LIBDIR
 	@. $(CONFIG); \
 	  set -x; \
 	  cd $$LIBDIR; \
-	  chmod 644 deutsch.aff deutsch.hash
+	  chmod 644 deutsch.aff deutschmed.hash deutschlxg.hash
+	  ln -sf deutschlxg.hash deutsch.hash; ln -sf deutschlxg.hash german.hash
 
-deutsch.hash:	$(BUILDHASH) $(AFFIXES) deutsch.dict
-	rm -f deutsch.hash
-	$(BUILDHASH) deutsch.dict $(AFFIXES) deutsch.hash
+deutschmed.hash:	$(BUILDHASH) $(AFFIXES) deutsch.med
+	rm -f deutschmed.hash
+	$(BUILDHASH) deutsch.med $(AFFIXES) deutschmed.hash
+
+deutschlxg.hash:	$(BUILDHASH) $(AFFIXES) deutsch.lxg
+	rm -f deutschlxg.hash
+	$(BUILDHASH) deutsch.lxg $(AFFIXES) deutschlxg.hash
 
-$(AFFIXES):	deutsch.7bit $(FIX8BIT)
+$(AFFIXES):	deutsch.aff $(FIX8BIT)
 	$(FIX8BIT) -8 < deutsch.7bit > $(AFFIXES)
 
+deutsch-isp.aff:	deutsch.7bit $(FIX8BIT)
+	$(FIX8BIT) -8 < deutsch.7bit > $(AFFIXES)
+		
 deutsch-alt.aff:	deutsch-alt.7bit $(FIX8BIT)
 	$(FIX8BIT) -8 < deutsch-alt.7bit > deutsch-alt.aff
 
 $(FIX8BIT):	../fix8bit.c
 	cd ..; $(MAKE) fix8bit
 
-deutsch.dict:	$(DICTALWAYS) $(DICTOPTIONS)
-	. $(CONFIG); \
-	  eval sort -f -o deutsch.dict $$MAKE_SORTTMP \
-	    $(DICTALWAYS) $(DICTOPTIONS)
+deutsch.med:	$(DICTALWAYS) README $(AFFIXES)
+	@. $(CONFIG); \
+	  set -x; \
+	  eval sort -u -t/ +0f -1 +0 $$MAKE_SORTTMP -o deutsch.med \
+	    $(DICTALWAYS)
 
+deutsch.lxg:	$(DICTALWAYS) $(DICTXLARGE) README $(AFFIXES)
+	@. $(CONFIG); \
+	  set -x; \
+	  eval sort -u -t/ +0f -1 +0 $$MAKE_SORTTMP -o deutsch.lxg \
+	    $(DICTALWAYS) $(DICTXLARGE)
+	
 #
 #	The following dependency can be executed when ispell is unpacked,
 #	to unpack the dictionaries.
@@ -191,7 +229,7 @@
 #	dictionary file.
 #
 dictclean:
-	rm -f deutsch.dict
+	rm -f deutsch.med deutsch.lxg
 
 #	required to be present in all other language Makefiles as
 #	well, even though it doesn't have to do anything in those
diff -Nur ispell-3.2.06.orig/languages/english/english.4l ispell-3.2.06/languages/english/english.4l
--- ispell-3.2.06.orig/languages/english/english.4l	Wed Jul 25 23:51:47 2001
+++ ispell-3.2.06/languages/english/english.4l	Thu Dec 26 23:26:21 2002
@@ -59,15 +59,15 @@
 .\"
 .TH ENGLISH 4
 .SH NAME
-english \- flag format for English \fIokspell\fP dictionaries
+english \- flag format for English \fIispell\fP dictionaries
 .SH DESCRIPTION
 English dictionaries for
-.IR okspell (1)
+.IR ispell (1)
 supports 3 prefix and 14 suffix flags.
 For a detailed description of how
-.I okspell
+.I ispell
 handles flags and capitalization, see
-.IR okspell (4).
+.IR ispell (4).
 This manual page only describes flags usable in dictionaries built
 using the
 .I english.aff
@@ -86,7 +86,7 @@
 .PP
 In the following list, an asterisk indicates that a flag participates in
 cross-product formation (see
-.IR okspell (4)).
+.IR ispell (4)).
 .PP
 The meaning of the prefix flags is as follows:
 .IP "*A"
@@ -286,4 +286,4 @@
 .fi
 .RE
 .SH "SEE ALSO"
-okspell(1), okspell(4)
+ispell(1), ispell(4)
diff -Nur ispell-3.2.06.orig/lookup.c ispell-3.2.06/lookup.c
--- ispell-3.2.06.orig/lookup.c	Wed Jul 25 23:51:46 2001
+++ ispell-3.2.06/lookup.c	Thu Dec 26 23:29:03 2002
@@ -88,6 +88,7 @@
 #include "ispell.h"
 #include "proto.h"
 #include "msgs.h"
+#include <unistd.h>
 
 int		linit P ((void));
 #ifdef INDEXDUMP
diff -Nur ispell-3.2.06.orig/tree.c ispell-3.2.06/tree.c
--- ispell-3.2.06.orig/tree.c	Wed Jul 25 23:51:46 2001
+++ ispell-3.2.06/tree.c	Thu Dec 26 23:29:47 2002
@@ -91,6 +91,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <unistd.h>
 #include "config.h"
 #include "ispell.h"
 #include "proto.h"
