--- ispell-3.4.06/Makefile.orig	2023-12-16 21:20:58.833790969 +0100
+++ ispell-3.4.06/Makefile	2023-12-16 21:21:27.686967991 +0100
@@ -336,7 +336,6 @@ install-basic:
 	@. ./config.sh; \
 	  set -x; \
 	  cd $(DESTDIR)$$BINDIR; \
-	  strip ispell$$EXEEXT; \
 	  chmod 755 ispell$$EXEEXT
 	@. ./config.sh; \
 	  set -x; \
@@ -392,7 +391,6 @@ install-dictbuild:
 	@. ./config.sh; \
 	  set -x; \
 	  cd $(DESTDIR)$$BINDIR; \
-	  strip buildhash$$EXEEXT icombine$$EXEEXT ijoin$$EXEEXT; \
 	  chmod 755 buildhash$$EXEEXT icombine$$EXEEXT ijoin$$EXEEXT \
 	    munchlist findaffix tryaffix
 	@. ./config.sh; \
