--- ispell-3.1/ispell.texinfo.orig	Wed Nov  2 19:44:23 1994
+++ ispell-3.1/ispell.texinfo	Sat Oct 13 22:18:02 2001
@@ -8,6 +8,11 @@
 @finalout
 @end iftex
 
+@dircategory Miscellaneous:
+@direntry
+* ispell: (ispell).			Interactive spelling checking.
+@end direntry
+
 @ifinfo
 This file documents ISPELL, an interactive spelling corrector.
 
@@ -73,23 +78,6 @@
 by the Foundation.
 
 @end titlepage
-@menu
-* Emacs::			Using ispell from emacs
-* Old Emacs::			Old Emacs
-* Private::			Your private dictionary
-* Command summary::		All commands in emacs mode
-* Near misses::			Definition of a near miss
-* History::			Where it came from
-
- --- The Detailed Node Listing ---
-
-Using ispell from emacs
-
-* Word::			Checking a single word
-* Buffer::			Checking a whole buffer
-* Region::			Checking a region
-* Multiple Dictionaries::	Using Multiple Dictionaries
-@end menu
 
 @node Top, Emacs, (dir), (dir)
 
@@ -198,7 +186,7 @@
 the shell.
 @end table
 
-Of course, you can also type @kbd{@ctrl{G}} to stop the command without
+Of course, you can also type @kbd{C-G} to stop the command without
 changing anything.
 
 If you make a change that you don't like, just use emacs' normal undo
@@ -215,7 +203,7 @@
 
 When you finish with one word, the cursor is automatically moved to the
 next.  If you want to stop in the middle of the list type @kbd{X} or
-@kbd{@ctrl{G}}.
+@kbd{C-G}.
 
 @node Region, Multiple Dictionaries, Buffer, Emacs
 @subsection Checking a region
--- ispell-3.1/Makefile.orig	Sat Oct 13 22:19:30 2001
+++ ispell-3.1/Makefile	Sat Oct 13 22:23:20 2001
@@ -323,7 +323,7 @@
 	  . ./config.sh; \
 	  set $(SHELLDEBUG); \
 	  if [ -d $$TEXINFODIR ]; then \
-	    if sh iwhich makeinfo >/dev/null \
+	    if which makeinfo >/dev/null \
 	      &&  makeinfo ispell.texinfo 2>/dev/null; then \
 		:; \
 	    elif sh iwhich $$EMACS >/dev/null; then \
