--- sq.c.old	Tue Jan 25 19:32:18 1994
+++ sq.c	Sun Dec 20 20:57:14 1998
@@ -75,15 +75,16 @@
     };
 
 #define MAX_PREFIX	(sizeof (size_encodings) - 1)
-
+#define MAX_WORD_LEN	257
+    
 int main (argc, argv)
     int		argc;
     char *	argv[];
     {
-    char	word[257];
-    static char	prev[257] = "";
+    char	word[MAX_WORD_LEN];
+    static char	prev[MAX_WORD_LEN] = "";
 
-    while (gets (word) != NULL)
+    while (fgets (word,sizeof(word),stdin) != NULL)
 	trunc (word, prev);
     return 0;
     }
 
--- unsq.c.old	Tue Jan 25 19:32:18 1994
+++ unsq.c	Sun Dec 20 20:59:36 1998
@@ -76,15 +76,16 @@
     };
 
 #define MAX_PREFIX	(sizeof (size_encodings) - 1)
-
+#define MAX_WORD_LEN	257
+    
 extern void	exit P ((int status));
 
 int main (argc, argv)
     int			argc;
     char *		argv[];
     {
-    char		word[257];
-    static char		prev[257] = "";
+    char		word[MAX_WORD_LEN];
+    static char		prev[MAX_WORD_LEN] = "";
 
     while (!expand (word, prev))
         puts (word);
@@ -116,7 +117,7 @@
     wordp = word;
     while (same_count--)
 	*wordp++ = (*prevp++);
-    if (gets (wordp) == NULL)
+    if (fgets (wordp,sizeof(wordp),stdin) == NULL)
 	{
 	(void) fprintf (stderr, UNSQ_C_SURPRISE_EOF);
 	exit (1);
--- ijoin.c.old	Mon Dec 21 01:14:47 1998
+++ ijoin.c	Mon Dec 21 01:15:23 1998
@@ -133,8 +133,6 @@
 static int		unpairable1 = 0; /* NZ if -a1 */
 static int		unpairable2 = 0; /* NZ if -a2 */
 
-extern int	strcmp ();
-
 int main (argc, argv)			/* Join files */
     int			argc;		/* Argument count */
     char *		argv[];		/* Argument vector */
